### PR TITLE
Improve pid file management

### PR DESCRIPTION
**Pull request by fluca1978**
These commits improve a little the management of the pid file by `main` application.

### CODE REVIEW

File: src/main.c [hunk 1]
```diff
@@ -1954,17 +1958,20 @@ create_pidfile(void)
 
    if (strlen(config->pidfile) > 0)
    {
-      // check pidfile is not there
-      if (access(config->pidfile, F_OK) == 0)
+      pid = getpid();
+
+      fd = open(config->pidfile, O_WRONLY | O_CREAT | O_EXCL, 0644);
```
Comment (jesperpedersen) [inline src/main.c:1963]: Can you change this to `0640` ?

**Review by jesperpedersen — CHANGES_REQUESTED** [review]

**Comment by jesperpedersen** [comment]
Just squash the commits with both issues referenced.

File: src/main.c (hunk 1 above)
Comment (fluca1978) [reply src/main.c:1969]: sure

**Comment by fluca1978** [comment]
> Just squash the commits with both issues referenced.

ok, but I like more to keep commits separated...

**Comment by jesperpedersen** [comment]
Could you split it into separate PRs then ? One for each issue - otherwise I can do it as part of the merge.

**Comment by fluca1978** [comment]
> Could you split it into separate PRs then ? One for each issue - otherwise I can do it as part of the merge.

I don't know how to split commits once I've rebased them and force pushed. I guess this is fine as a single PR.

My personal idea is to keep related commits into a PR/branch, and commits as short as possible to avoid "jumbo patches".

**Comment by jesperpedersen** [comment]
Merged.

Thanks for your contribution !

**Pull request closed without merging**